### DISCUSSION TITLE
storage policy and schema mapping evolution

### DISCUSSION THREAD

**Discussion by baallan** [ovis-hpc/ldms#878] [Ideas]
Here we will capture use cases and design thoughts.

**Comment by nichamon** [comment]
User's Group meeting (Feb 14, 2022)

**Suggestions & Ideas**
- @valleydlr wants to have the description of each column stored somewhere that users can query on, e.g., "ts" comes from the set timestamp, 'rate' is calculated by (A+B)/C.
- @morrone wants to be able to tell storage plugins the data type, e.g., lists.
- @tom95858 suggested that, in the future, we could specify simple math manipulation in storage policy, e.g., { "src": (A + B)/C, "dst" : "rate" }.
- We decided to make decomposition code as plugins. Initially, we will have 'static' and 'as_is'.
- @morrone suggested having a keyword to specify values when a metric value is missing. @narategithub suggested we can use `'fill' : "value"`.
